### PR TITLE
Fix Coverity issues

### PR DESCRIPTION
**Pull request by mikisch81**
* Report [before ](http://ta-lab-master.kfn.arm.com:8080/job/uVisor/job/Coverity/40/HTML_Report/) this PR has 91 issues.
* Report [with](http://ta-lab-master.kfn.arm.com:8080/job/uVisor/job/Coverity/41/HTML_Report/) this PR has 85 issues.

Issues resolved: 24-26, 78, 81, 82

@Patater @alzix @danny4478 @romkuz01

### CODE REVIEW

**Comment by mikisch81** [comment]
This is based on #507 so it should be merged only afterwards.

File: core/system/src/context.c [hunk 1]
```diff
@@ -75,7 +75,7 @@ static void context_state_push(TContextSwitchType context_type, uint8_t src_id,
 static TContextPreviousState * context_state_pop(void)
 {
     /* Check that the state stack does not underflow. */
-    if (!g_context_p) {
+    if (0 >= g_context_p) {
```
Comment (Patater) [inline core/system/src/context.c]: Should this be `g_context_p == 0`?

File: core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c [hunk 2]
```diff
@@ -138,7 +138,7 @@ uint8_t vmpu_region_bits(uint32_t size)
 {
     uint8_t bits;
 
-    bits = vmpu_bits(size) - 1;
+    bits = (0 == size) ? 0: (vmpu_bits(size) - 1);
```
Comment (Patater) [inline core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c]: Do we want vmpu_region_bits to return 0 for size 0? It's getting rounded up from 0 to 1, as currently written. Probably we should just consider size 0 as a special case, perhaps asserting that size is not 0.

File: core/system/src/context.c (hunk 1 above)
Comment (mikisch81) [reply core/system/src/context.c]: yes

File: core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c (hunk 2 above)
Comment (mikisch81) [reply core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c]: I'm in favor of asserting, there is no reason size should be 0.

**Comment by mikisch81** [comment]
Can someone besides @Patater take a look?
@alzix @danny4478 @romkuz01

File: core/system/src/context.c [hunk 3]
```diff
@@ -87,7 +87,7 @@ static TContextPreviousState * context_state_pop(void)
 /* Return a pointer to the previous box context state. */
 TContextPreviousState * context_state_previous(void)
 {
-    if (!g_context_p) {
+    if (0 >= g_context_p) {
```
Comment (alzix) [inline core/system/src/context.c]: please explain why the very same expression in lines 78 and 90 was replaced by different conditions?

Comment (mikisch81) [reply core/system/src/context.c]: It was forgotten..

**Comment by Patater** [comment]
Is the new Coverity report available?

**Comment by mikisch81** [comment]
@patater It is best to run the analysis after #507 is merged.

File: core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c [hunk 4]
```diff
@@ -136,9 +136,9 @@ uint32_t vmpu_round_up_region(uint32_t addr, uint32_t size)
 
 uint8_t vmpu_region_bits(uint32_t size)
 {
-    uint8_t bits;
+    assert(0 != size);
 
-    bits = vmpu_bits(size) - 1;
+    uint8_t bits = vmpu_bits(size) - 1;
```
Comment (Patater) [inline core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c]: I'm particularly curious if this change is enough to make Coverity happy about this case.

**Comment by Patater** [comment]
@mikisch81 Can we verify that we are actually fixing issues with this change, at least?

**Comment by mikisch81** [comment]
@Patater As this PR depends on #507, I need to do some tweaking to the Jenkins job, but I can do it.

**Comment by mikisch81** [comment]
@Patater 
This was tested with my own forks

* Without this PR: [144 issues](http://ta-lab-master.kfn.arm.com:8080/job/uVisor/job/Coverity/74/HTML_Report/)
* With this PR: [139 issues](http://ta-lab-master.kfn.arm.com:8080/job/uVisor/job/Coverity/74/HTML_Report/)

**Comment by Patater** [comment]
@mikisch81 Does that mean this PR fixes all the issues you intended to fix? Or are some of our fixes insufficient after addressing review feedback?

**Comment by mikisch81** [comment]
@Patater I need to further investigate the reports for that.
It will be much easier after #507 is merged because I am always changing the Jenkins job to test different commits.

**Comment by Patater** [comment]
@mikisch81 OK. #507 is almost there. One remaining question on my end.

**Comment by Patater** [comment]
#507 is merged now.

**Comment by mikisch81** [comment]
### Updated reports
* Without this PR: [141 issues](http://ta-lab-master.kfn.arm.com:8080/job/uVisor/job/Coverity/77/HTML_Report/)
* Without this PR: [136 issues](http://ta-lab-master.kfn.arm.com:8080/job/uVisor/job/Coverity/78/HTML_Report/)

File: core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c (hunk 4 above)
Comment (Patater) [reply core/vmpu/src/mpu_armv7m/vmpu_armv7m_mpu.c]: Coverity is no longer complaining about this. Nice.

**Comment by mikisch81** [comment]
@Patater 
Looks like the CI is currently not stable.

**Comment by Patater** [comment]
@mikisch81 Yeah, looks like the `git clone` is taking too long (more than 10 minutes).

**Comment by mikisch81** [comment]
@patater this affects all the PRs

**Comment by Patater** [comment]
Please squash commits as appropriate.

**Comment by Patater** [comment]
@alzix LGTY?

**Comment by alzix** [comment]
LGTM

**Pull request merged**